freud/diffraction.pyx

```diff
@@ -0,0 +1,348 @@
+        ----------
+        grid_size : int, size of the diffraction grid (default 512)
+        zoom : (default 1)
+        peak_width : (default 1)
+        length_scale : (default 3.905)
+        bot : (default 4e-6)
+        top : (default 0.7)
+        """
+        self.N = grid_size
+        self.zoom = zoom
+        self.peak_width = peak_width
+        self.bin_w = 2.0
+        self.length_scale = length_scale
+        self.bot = bot
+        self.top = top
+
+    def load(self, system):
+        """
+        def load(self, xyz, L):
+            Loads the particle positions and box dimensions for diffraction.
+            Note: only supports orthorhombic boxes
+            Parameters
+            ----------
+            xyz: np.ndarray (N,3), positions of each particle
+            L: iterable object, lengths of box vectors
+            self.box = np.array(
+                    [[L[0], 0.0, 0.0],
+                     [0.0, L[1], 0.0],
+                     [0.0, 0.0, L[2]]]
+                    )
+            self.orig = np.copy(xyz)
+            self.orig, self.image =
+                pbc.shift_pbc(xyz, np.array([L[0], L[1], L[2]]))
+        """
+        box, positions = system
+        self.box = box.to_matrix()
+        self.orig = np.copy(positions)
+
+    def pbc_2d(self, xy, N):
+        """
+        Reasonably fast periodic boundary conditions in two dimensions.
+        Normalizes xy coordinates to the grid size, N.
+        Parameters
+        ----------
+        xy : numpy.ndarray (N,2), cartesian coordinates from [-0.5, 0.5)
+        to be mapped to [0, N)
+        N : int, grid size
+        Returns
+        -------
+        numpy.ndarray (N,2), particle bins indices
+        in the x and y directions.
+        """
+        xy -= np.rint(xy) - 0.5
+        xy *= N
+        xy %= N
+        return xy.astype(int)
+
+    def bin(self, xy, N):
+        """
+        Quickly counts intensities for particles on 2D grid.
+        Parameters
+        ----------
+        xy : numpy.ndarray (N,2), array of bin indices
+        N : int, grid size
+        Returns
+        -------
+        im : numpy.ndarray (N,N), grid of intensities.
+        """
+        t = xy.view(np.dtype((np.void, xy.dtype.itemsize * xy.shape[1])))
+        _, ids, counts = np.unique(t, return_index=True, return_counts=True)
+        unique_xy = xy[ids]
+        N = int(N)
+        im = np.zeros((N, N))
+        for x, c in zip(unique_xy, counts):
+            im[x[1], x[0]] = c
+        return im
+
+    def calc_proj(self, rot):
+        """
+        TODO
+        Note: orthorhombic boxes only
+        Parameters
+        ----------
+        rot : numpy.ndarray (3,3), rotation matrix
+        Returns
+        -------
+        numpy.ndarray (2,2), inverse shear matrix
+        """
+        # s = np.dot(rot.T, self.box)  # rotated box vectors
+        s = rowan.rotate(rot, self.box)
+        xy = np.absolute(s[0, 0] * s[1, 1] - s[0, 1] * s[1, 0])
```

Comment:
These shear factors are given directly by the freud Box object. Verify the math and use those instead.